fe/fe-core/src/main/java/com/starrocks/credential/hdfs/HDFSCloudConfiguration.java

```diff
@@ -14,38 +14,85 @@
 import com.starrocks.thrift.TCloudConfiguration;
+import com.starrocks.thrift.TCloudType;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.starrocks.credential.CloudConfigurationConstants.HDFS_CONFIG_RESOURCES;
+import static com.starrocks.credential.CloudConfigurationConstants.HDFS_RUNTIME_JARS;
 
 public class HDFSCloudConfiguration implements CloudConfiguration {
+    private static final Logger LOG = LogManager.getLogger(HDFSCloudConfiguration.class);
+
     private final HDFSCloudCredential hdfsCloudCredential;
+    private String configResources;
+    private String runtimeJars;
+
+    private static final String CONFIG_RESOURCES_SEPERATOR = ",";
 
     public HDFSCloudConfiguration(HDFSCloudCredential hdfsCloudCredential) {
         Preconditions.checkNotNull(hdfsCloudCredential);
         this.hdfsCloudCredential = hdfsCloudCredential;
     }
 
+    public void setConfigResources(String configResources) {
+        this.configResources = configResources;
+    }
+
+    public void setRuntimeJars(String runtimeJars) {
+        this.runtimeJars = runtimeJars;
+    }
+
     public HDFSCloudCredential getHdfsCloudCredential() {
         return hdfsCloudCredential;
     }
 
+    public void addConfigResourcesToConfiguration(String configResources, Configuration conf) {
+        if (Strings.isNullOrEmpty(configResources)) {
+            return;
+        }
+        String[] parts = configResources.split(CONFIG_RESOURCES_SEPERATOR);
+        for (String p : parts) {
+            Path path = new Path(StarRocksFE.STARROCKS_HOME_DIR + "/conf/", p);
+            LOG.debug(String.format("Add path '%s' to configuration", path.toString()));
+            conf.addResource(path);
+        }
+    }
+
     @Override
     public void toThrift(TCloudConfiguration tCloudConfiguration) {
-        // TODO
+        tCloudConfiguration.setCloud_type(TCloudType.HDFS);
+        Map<String, String> properties = new HashMap<>();
+        hdfsCloudCredential.toThrift(properties);
+        properties.put(HDFS_CONFIG_RESOURCES, configResources);
+        properties.put(HDFS_RUNTIME_JARS, runtimeJars);
+        tCloudConfiguration.setCloud_properties_v2(properties);
     }
 
     @Override
     public void applyToConfiguration(Configuration configuration) {
-        // TODO
+        hdfsCloudCredential.applyToConfiguration(configuration);
+        addConfigResourcesToConfiguration(configResources, configuration);
     }
 
     @Override
     public String getCredentialString() {
-        return hdfsCloudCredential.getCredentialString();
+        return "HDFSCloudConfiguration{" +
+                "configResources=" + configResources +
+                ", runtimeJars=" + runtimeJars +
+                ", credential=" + hdfsCloudCredential.toString() + "}";
```

Comment:
it's `getCredentialString()`, not `toString()`

Reply:
fixed.